R/createAttributePlan.R

```diff
@@ -0,0 +1,190 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+##---------------------------------------------------------------
+##                    Create attribute plan                    --
+##---------------------------------------------------------------
+#' @param jaspResults {object} Object that will contain all results from the analysis and connect it to the output.
+#' @param dataset {object} (optional) tabular data, if available for the analysis.
+#' @param options {list} A named list of interface options selected by the user.
+#' @seealso .findPlan()
+##---------------------------------------------------------------
+CreateAttributePlan <- function(jaspResults, dataset = NULL, options, ...) {
+  # Constraints to create a plan
+  risk_vars <- c("aql", "prod_risk", "rql", "cons_risk")
+  # Quality levels
+  pd_vars <- c("pd_lower", "pd_upper", "pd_step")
+  depend_vars <- c(pd_vars, risk_vars, "lotSize", "distribution")
+
+  # Check if the container already exists. Create it if it doesn't.
+  if (is.null(jaspResults[["createContainer"]]) || jaspResults[["createContainer"]]$getError()) {
+    createContainer <- createJaspContainer(title = "")
+    createContainer$dependOn(depend_vars)
+    jaspResults[["createContainer"]] <- createContainer
+  } else {
+    createContainer <- jaspResults[["createContainer"]]
+  }
+
+  # Plan table outline
+  plan_table <- createJaspTable(title = "Generated Sampling Plan")
+  plan_table$addColumnInfo(name = "col_1", title = "", type = "string")
+  plan_table$addColumnInfo(name = "col_2", title = "Value", type = "integer")
+  plan_table[["col_1"]] <- c("Sample size", "Acceptance number")
+  plan_table$position <- 1
+  createContainer[["findPlanTable"]] <- plan_table
+
+  # Probability table outline
+  prob_table <- createJaspTable(title = "Acceptance probabilities at AQL and RQL")
+  prob_table$addColumnInfo(name = "col_1", title = "", type = "string")
+  prob_table$addColumnInfo(name = "col_2", title = "Proportion Non-conforming", type = "number")
+  prob_table$addColumnInfo(name = "col_3", title = "Acceptance Probability", type = "number")
+  prob_table$addColumnInfo(name = "col_4", title = "Rejection Probability", type = "number")
+  prob_table[["col_1"]] <- c("AQL", "RQL")
+  prob_table$position <- 2
+  createContainer[["findProbTable"]] <- prob_table
+  
+  # Error handling for hypergeometric distribution
+  aql <- round(options$aql, 3)
+  rql <- round(options$rql, 3)
```

Comment:
why are you rounding these values? If this is for displaying the values in a table, you don't need to do this yourself (there's a general option for that).

Reply:
I need to do some comparisons between these numbers, and I found that without rounding off, the floating point arithmetic sometimes gives unexpected results. 
Another solution could also be to do any such comparisons (aql >= rql, etc.) with some pre-specified tolerance.
Let me know which approach you prefer. (For the purpose of results, 3 decimals is enough here.)

Reply:
I'd use `isTRUE(all.equal(a, b))`, which uses the default tolerance of `sqrt(.Machine$double.eps)`, `1.490116e-08` on my machine.

Reply:
Removed the rounding off.